### PR TITLE
Feature/sg 119 cooldown epochs

### PR DESCRIPTION
**Pull request by shaydeci**
Learning rate cool down epochs support (i.e continuing training with the last scheduled learning rate).
Additionally I refactored our lr callbacks code that had a lot of duplicated code fragments.

### CODE REVIEW

File: src/super_gradients/training/utils/callbacks.py [hunk 1]
```diff
@@ -237,7 +237,25 @@ def __init__(self, phase, initial_lr, update_param_groups, train_loader_len, net
         self.net = net
         self.training_params = training_params
 
-    def __call__(self, *args, **kwargs):
+    def __call__(self, context: PhaseContext, **kwargs):
+        if self.enable_lr_scheduling(context):
+            self.perform_scheduling(context)
+
+    def enable_lr_scheduling(self, context: PhaseContext):
```
Comment (ofrimasad) [inline src/super_gradients/training/utils/callbacks.py:244]: the naming sounds like an action (like you enable something)
maybe change it to something like `is_lr_scheduling_enabled`

File: src/super_gradients/training/utils/callbacks.py [hunk 2]
```diff
@@ -342,16 +370,16 @@ def __init__(self, max_epochs, **kwargs):
         super(PolyLRCallback, self).__init__(Phase.TRAIN_BATCH_STEP, **kwargs)
         self.max_epochs = max_epochs
 
-    def __call__(self, context: PhaseContext):
-        # POLYNOMIAL LEARNING RATE
-        if self.training_params.lr_warmup_epochs <= context.epoch:
-            effective_epoch = context.epoch - self.training_params.lr_warmup_epochs
-            effective_max_epochs = self.max_epochs - self.training_params.lr_warmup_epochs
+    def perform_scheduling(self, context):
+        effective_epoch = context.epoch - self.training_params.lr_warmup_epochs - self.training_params.lr_cooldown_epochs
```
Comment (ofrimasad) [inline src/super_gradients/training/utils/callbacks.py:374]: why do you reduce lr_cooldown_epochs here?

**Review by ofrimasad — COMMENTED** [review]
minor comments inline

File: src/super_gradients/training/utils/callbacks.py (hunk 2 above)
Comment (shaydeci) [reply src/super_gradients/training/utils/callbacks.py]: fixed

File: src/super_gradients/training/utils/callbacks.py (hunk 1 above)
Comment (shaydeci) [reply src/super_gradients/training/utils/callbacks.py]: done.

**Review by ofrimasad — APPROVED** [review]
LGTM
